server/filestore.go

```diff
@@ -3933,12 +3934,13 @@ func (fs *fileStore) hashKeyForBlock(index uint32) []byte {
 
 func (mb *msgBlock) setupWriteCache(buf []byte) {
 	// Make sure we have a cache setup.
-	if mb.cache != nil {
+	if mb.cache.Value() != nil {
 		return
 	}
 
 	// Setup simple cache.
-	mb.cache = &cache{buf: buf}
+	mbcache := &cache{buf: buf}
+	mb.cache = elastic.Make(mbcache)
```

Comment:
Why not one line?

Reply:
Because we go on to use `mbcache` further down.